src/DIPS.Xamarin.UI/Controls/Toast/Toast.cs

```diff
@@ -0,0 +1,274 @@
+        private Dictionary<string, Grid> ToastContainers { get; } = new Dictionary<string, Grid>();
+        private ToastLayout ToastLayout { get; set; }
+        private ToastOptions ToastOptions { get; set; }
+
+        private void OnPageDisappearing(object sender, Page page)
+        {
+            if (page is ContentPage contentPage)
+            {
+                CancellationSource.Cancel();
+                _ = HideToast(contentPage);
+            }
+        }
+
+        private void OnPageAppearing(object sender, Page e)
+        {
+            Initialize();
+        }
+
+        /// <summary>
+        ///     Initialize Toast control.
+        /// </summary>
+        public void Initialize()
+        {
+            _ = GetToastContainerSettingUpIfNeededAsync();
+        }
+
+        /// <summary>
+        ///     Displays the Toast control
+        /// </summary>
+        /// <param name="text">Text to be displayed in the Toast control</param>
+        /// <param name="options">An <see cref="Action{ToastOptions}" /> to modify Toast options</param>
+        /// <param name="layout">An <see cref="Action{ToastLayout}" /> to modify Toast layout</param>
+        /// <returns>A void <c>Task</c></returns>
+        public async Task DisplayToast(string text, Action<ToastOptions> options, Action<ToastLayout> layout)
+        {
+            var toastOptions = new ToastOptions();
+            options(toastOptions);
+            var layoutOptions = new ToastLayout();
+            layout(layoutOptions);
+
+            await DisplayToast(text, toastOptions, layoutOptions);
+        }
+
+        /// <summary>
+        ///     Displays the Toast control
+        /// </summary>
+        /// <param name="text">Text to be displayed in the Toast control</param>
+        /// <param name="options"><see cref="ToastOptions" /> to set for the Toast control</param>
+        /// <param name="layout"><see cref="ToastLayout" /> to set for the Toast control</param>
+        /// <returns>A void <c>Task</c></returns>
+        public async Task DisplayToast(string text, ToastOptions options = null, ToastLayout layout = null)
+        {
+            // set properties
+            ToastOptions = options ?? new ToastOptions();
+            ToastLayout = layout ?? new ToastLayout();
+
+            // get toast container
+            var toastContainer = await GetToastContainerSettingUpIfNeededAsync();
+            if (toastContainer == null)
+            {
+                return;
+            }
+
+            // toast view
+            var toastView = GetToast(text);
+            toastContainer.Children.Add(toastView);
+
+            // animate toast
+            if (ToastOptions.DisplayAnimation != null)
+            {
+                await ToastOptions.DisplayAnimation(toastView);
+            }
+
+            // hide toast
+            if (ToastOptions.Duration > 0)
```

Comment:
What if Duration is <= 0?
Toast is never hidden?

Reply:
See my comment here: https://github.com/DIPSAS/DIPS.Xamarin.UI/pull/195#discussion_r464814266

Reply:
@Illedan is asking if it will never be hidden if the `Duration` is <= 0, yes

it's documented so the user can know it, for use cases like letting the user Tap on the Toast like in Twitter. This would be more ideal than forcing the user to set a larger value to prevent it from closing automatically.

Reply:
I don't agree with 0 meaning it won't hide, as I said in the comment I linked to.

Reply:
For me this is a preferential request, `0` is defined in the code as it will stop hiding. 

If the animation is removed and the time is set to `0`, the user won't see the Toast if it's meant to close when `0`. 

The behaviour for `<=0` is the very same if the Toast is not meant to close on those values, even with or without animation.

Reply:
However I will move to using `<= -1` to get this merged